libmscore/rest.h

```diff
@@ -93,10 +94,12 @@ class Rest : public ChordRest {
       virtual QPointF stemPosBeam() const;
 
       virtual void localSpatiumChanged(qreal oldValue, qreal newValue) override;
+      virtual QVariant propertyDefault(Pid) const override;
+      void resetProperty(Pid id);
+      Sid getPropertyStyle(Pid pid) const;
```

Comment:
Shouldn't this be private and also marked virtual & override?  Aside from that, and the test failures, looks good to me

Reply:
Probably, yes. For some reason I can't see the log for the failed test.

Reply:
The tests fail because of an apparent segmentation violation highlighted by Address Sanitizer, during operations such as joinmeasure, splitmeasure or rewritemeasure; here is an example stack trace:
`==18320==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x000000c6641c sp 0x7ffc5e90bb70 bp 0x7ffc5e90bb90 T0)
#0 0xc6641b in Ms::Score::styleV(Ms::Sid) const libmscore/score.h:866
#1 0x13603df in Ms::ScoreElement::styleValue(Ms::Pid, Ms::Sid) const libmscore/scoreElement.cpp:898
#2 0x1356bfa in Ms::ScoreElement::initElementStyle(std::vector<Ms::StyledProperty, std::allocator<Ms::StyledProperty> > const*) libmscore/scoreElement.cpp:212
#3 0xed4193 in Ms::Rest::Rest(Ms::Score*) libmscore/rest.cpp:52
#4 0x12d6042 in Ms::TrackList::appendGap(Ms::Fraction const&) libmscore/range.cpp:225
#5 0x12d735a in Ms::TrackList::read(Ms::Segment const*, Ms::Segment const*) libmscore/range.cpp:310
#6 0x12dd6c0 in Ms::ScoreRange::read(Ms::Segment*, Ms::Segment*, bool) libmscore/range.cpp:665
#7 0xb8516f in Ms::Score::rewriteMeasures(Ms::Measure*, Ms::Measure*, Ms::Fraction const&, int) libmscore/edit.cpp:490
#8 0xb870ea in Ms::Score::rewriteMeasures(Ms::Measure*, Ms::Fraction const&, int) libmscore/edit.cpp:608
#9 0xb8a174 in Ms::Score::cmdAddTimeSig(Ms::Measure*, int, Ms::TimeSig*, bool) libmscore/edit.cpp:832
#10 0x77ccf4 in TestClef::clef2() mtest/libmscore/clef/tst_clef.cpp:72`

Reply:
Unfortunately, I don't know what this means.

Reply:
It can be reproduced from scratch: from initial score, select two (rest-filled) measures and join them (Tools → Measure → Join Measures); in my debug build this causes a crash.
At scoreElement.cpp line 212 `setProperty(spp.pid, styleValue(spp.pid, getPropertyStyle(spp.pid)));` is called for property Ms::Sid::yPosMMRestNumber, this results in going to line 898 of scoreElement.cpp, `return score()->styleV(sid);` but apparently for this rest (which I think is the one being added) the score is still a null pointer.

Reply:
Thanks, now I understand how to read the stack. So the fix might be to change range.cpp line 225, where it initializes a rest with `0`, to simply pass it the expected `score` pointer?

Reply:
Of course, now I can't figure out how to get that `Score` pointer in range.cpp...

Reply:
Ah, a couple of `if` statements took care of it. Update about to be pushed.

Reply:
All checks have passed.